### ISSUE TITLE
Maximizing anonymity

### ISSUE THREAD

**Issue by nick** [kendricktan/heiswap-dapp#3]
Given the current ring size of 5, does that mean anonymity is limited to 1 of those 5 accounts? Can a given withdrawal tx be traced to 1 of the 5 depositors?

Would increasing the ring size to 100 and disabling the ability to close the ring early increase anonymity?

If a depositor takes the first slot in a ring, could an attacker simply fill up the other slots, watch for a withdrawal event and therefore attribute it to the original depositor?

**Comment by kendricktan** [comment]
Yep, a ring size of 5 means there is a 1/5 chance of finding out the link between the sender and receiver. However, increasing it to 100 makes it impossible to execute GAS-wise.

One way to avoid this would to be to continually go through the ring N times (or do it automatically in the smart contract) so the anonymity set increases to `1/(5^N)`.

That is one possible attack vector, and one way to reduce the chances of that happening would be to randomly allocate people into rings instead of allocating them sequentially :smiley:
